samza-azure/src/test/java/org/apache/samza/system/azureblob/avro/TestAzureBlobAvroWriter.java

```diff
@@ -0,0 +1,452 @@
+    int numberOfMessages = 10;
+    azureBlobAvroWriter.write(ome);
+    for (int i = 0; i < numberOfMessages; ++i) {
+      azureBlobAvroWriter.write(omeEncoded);
+    }
+    verify(mockDataFileWriter).appendEncoded(ByteBuffer.wrap(encodedRecord));
+    verify(mockDataFileWriter, times(numberOfMessages)).appendEncoded(ByteBuffer.wrap((byte[]) omeEncoded.getMessage()));
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void testWriteByteArrayWithoutSchema() throws Exception {
+    azureBlobAvroWriter =
+        spy(new AzureBlobAvroWriter(PowerMockito.mock(BlobContainerAsyncClient.class), mock(AzureBlobWriterMetrics.class),
+            threadPool, THRESHOLD, 60000, "test",
+            null, null, null, 1000, 100));
+    OutgoingMessageEnvelope omeEncoded = new OutgoingMessageEnvelope(new SystemStream(SYSTEM_NAME, "Topic1"), new byte[100]);
+    azureBlobAvroWriter.write(omeEncoded);
+  }
+
+  @Test(expected = IOException.class)
+  public void testWriteWhenDataFileWriterFails() throws Exception {
+    doThrow(new IOException("Failed")).when(mockDataFileWriter).appendEncoded(ByteBuffer.wrap(encodedRecord));
+    azureBlobAvroWriter.write(ome);
+  }
+
+  @Test
+  public void testClose() throws Exception {
+    azureBlobAvroWriter.close();
+    verify(mockDataFileWriter).close();
+    verify(mockAzureBlobOutputStream).close();
+  }
+
+  @Test(expected = SamzaException.class)
+  public void testCloseWhenDataFileWriterFails() throws Exception {
+    doThrow(new IOException("Failed")).when(mockDataFileWriter).close();
+
+    azureBlobAvroWriter.flush();
+    azureBlobAvroWriter.close();
+  }
+
+  @Test(expected = RuntimeException.class)
+  public void testCloseWhenOutputStreamFails() throws Exception {
+    doThrow(new RuntimeException("failed")).when(mockAzureBlobOutputStream).close();
+
+    azureBlobAvroWriter.close();
+  }
+
+  @Test
+  public void testFlush() throws Exception {
+    azureBlobAvroWriter.flush();
+    verify(mockDataFileWriter).flush();
+  }
+
+  @Test(expected = IOException.class)
+  public void testFlushWhenDataFileWriterFails() throws Exception {
+    doThrow(new IOException("Failed")).when(mockDataFileWriter).flush();
+    azureBlobAvroWriter.flush();
+  }
+
+  @Test
+  public void testMaxBlobSizeExceeded() throws Exception {
+    String blobUrlPrefix = "test";
+    long maxBlobSize = 1000;
+    BlobContainerAsyncClient mockContainerClient = PowerMockito.mock(BlobContainerAsyncClient.class);
+    azureBlobAvroWriter = spy(new AzureBlobAvroWriter(mockContainerClient,
+        mock(AzureBlobWriterMetrics.class), threadPool, THRESHOLD, 60000, blobUrlPrefix,
+        null, null, null, maxBlobSize, 10));
+
+    when(mockAzureBlobOutputStream.getSize()).thenReturn((long) maxBlobSize - 1);
+    PowerMockito.whenNew(AzureBlobOutputStream.class).withAnyArguments().thenReturn(mockAzureBlobOutputStream);
+    PowerMockito.whenNew(DataFileWriter.class).withAnyArguments().thenReturn(mockDataFileWriter);
```

Comment:
It looks like this currently just tests that multiple blob clients are created.
Would it be useful/feasible to test that the data gets appended to the correct places?